README.md

```diff
@@ -30,27 +51,249 @@ npm install --save ipfsd-ctl
+    })
+  })
+})
+```
+
+It's also possible to start the server from `.aegir` `pre` and `post` hooks. 
+
+```js
+'use strict'
+
+const createServer = require('./src').createServer
+
+const server = createServer()
+module.exports = {
+  karma: {
+    files: [{
+      pattern: 'test/fixtures/**/*',
+      watched: false,
+      served: true,
+      included: false
+    }],
+    singleRun: true
+  },
+  hooks: {
+    browser: {
+      pre: server.start.bind(server),
+      post: server.stop.bind(server)
+    }
+  }
+}
+```
+
+## Disposable vs non Disposable nodes
+
+`ipfsd-ctl` can create two types of node controllers, `disposable` and `non-disposable`. A disposable node will be created on a temporary repo which will be optionally initialized and started (the default), as well cleaned up on process exit. A non-disposable node on the other hand, requires the user to initialize and start the node, as well as stop and cleanup after wards. Additionally, a non-disposable will allow you to pass a custom repo using the `repoPath` option, if the `repoPath` is not defined, it will use the default repo for the node type (`$HOME/.ipfs` or `$HOME/.jsipfs`). The `repoPath` parameter is ignored for disposable nodes, as there is a risk of deleting a live repo.
+
+## IPFS executable types
+
+`ipfsd-ctl` allows spawning different types of executables, such as:
+
+> `go`
+
+Invoking `df.spawn({type: 'go', exec: '<path to go executable or empty to allow ipfsd to find one automatically>'})` will spawn a `go-ipfs` node.
+
+> `js`
+
+Invoking `df.spawn({type: 'js', exec: '<path to js executable or empty to allow ipfsd to find one automatically>'})` will spawn a `js-ipfs` node.
+
+> `proc`
+
+Invoking `df.spawn({type: 'proc', exec: require('ipfs')})` will spawn an `in-process-ipfs` node using the provided code reference that implements the core IPFS API. Note that, `exec` is required if `type: 'proc'` is used.
+
+### IPFS executables
+
+`ipfsd-ctl` no longer installs go-ipfs nor js-ipfs dependencies, instead it expects them to be provided by the parent project. In order to be able to use both go and js daemons, please make sure that your project includes these two npm packages as dependencies.
+
+- `ipfs` - the js-ipfs implementation
+- `go-ipfs-dep` - the packaged go-ipfs implementation
+
+## API
+
+### Daemon Factory
+
+#### Create a `DaemonFactory`
+
+> `DaemonFactory.create([options])` create a factory that will expose the `df.spawn` method
+
+- `options` - an optional object with the following properties
+  - `remote` bool - indicates if the factory should spawn local or remote nodes. By default, local nodes are spawned in Node.js and remote nodes are spawned in Browser environments.
+  - `port` number - the port number to use for the remote factory. It should match the port on which `DaemonFactory.server` was started. Defaults to 9999.
+
+> `DaemonFactory.createServer` create an instance of the bundled HTTP server used by the remote controller.
```

Comment:
The `DaemonFactory.createServer` should get its own section.

Reply:
Title can be "Create a `DaemonFactory Endpoint` - `const server = DaemonFactory.createServer([options])`"